### PR TITLE
Separate web server from main application logic

### PR DESCRIPTION
**Pull request by taiki-e**

### CODE REVIEW

File: src/server.rs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone, Default)]
+pub struct RobotOrigin {
+    pub position: [f32; 3],
+    pub quaternion: [f32; 4],
+}
+
+#[derive(Debug)]
+pub struct ServerData {
```
Comment (OTL) [inline src/server.rs:17]: `ServerData` sounds like some configurations for the webserver for me.
How about using `RobotState`, `RobotStateData` or `RobotServerData` ...?

Comment (OTL) [reply src/server.rs:17]: `ServerRobotData`?

File: src/app.rs [hunk 2]
```diff
@@ -453,8 +442,7 @@ static ABORTED: AtomicBool = AtomicBool::new(false);
 
 struct AppState {
     app: UrdfViewerApp,
-    #[cfg(not(target_arch = "wasm32"))]
-    data: Arc<Data>,
+    data: Arc<ServerData>,
```
Comment (OTL) [inline src/app.rs:445]: `data` has almost no meaning.

File: src/server.rs (hunk 1 above)
Comment (taiki-e) [reply src/server.rs:17]: From the user's point of view, I think `RobotStateHandle` would describe its nature exactly.

Comment (OTL) [reply src/server.rs:17]: I see, great!

Comment (taiki-e) [reply src/server.rs]: renamed to RobotStateHandle

File: src/app.rs (hunk 2 above)
Comment (taiki-e) [reply src/app.rs]: renamed to handle

**Review by OTL — APPROVED** [review]

**Pull request merged**